File: python/game/ui/cafeteria.py
from ..popup import Popup


def build_popup(h):
    s = h['state']
    p = Popup("Cafeteria")
    p.add_line("Grab a meal and recharge.")

    def get_coupon():
        if not s.flags['mealCoupon']:
            s.flags['mealCoupon'] = True
            h['addItem']('Meal Coupon')
            h['toast']('Meal coupon collected')
            h['rebuild']()

    def eat():
        if not s.flags['ateMeal'] and s.flags['mealCoupon']:
            s.flags['ateMeal'] = True
            h['setEnergy'](100)
            h['addXP'](5)
            h['addStress'](-5)  # Eating reduces stress
            h['addReputation'](2)  # Socializing at cafeteria
            h['completeQuest']('eatMeal')
            h['toast']('Meal eaten (+5 XP, Energy 100, -5 Stress, +2 Reputation)')
            h['rebuild']()
        elif s.flags['mealCoupon'] and s.energy < 80:
            # Can eat again if energy is low (but no quest completion)
            h['setEnergy'](min(100, s.energy + 30))
            h['addStress'](-3)
            h['addReputation'](1)
            h['toast']('Meal eaten (Energy +30, -3 Stress, +1 Reputation)')

    p.add_button(
        "Get Meal Coupon",
        get_coupon,
        primary=False,
        disabled=s.flags['mealCoupon']
    )
    p.add_button(
        "Eat Meal (+5 XP, Energy 100)",
        eat,
        primary=True,
        disabled=s.flags['ateMeal'] or not s.flags['mealCoupon']
    )
    return p

File: src/ui/cafeteria.js
(function(){
  window.UI = window.UI || {};
  window.UI.cafeteria = function(container, state, {addXP, addItem, completeQuest, setEnergy}){
    const root = document.createElement('div');
    root.className = 'ui-grid';

    const services = document.createElement('div');
    services.className = 'ui-card';
    const t1 = document.createElement('div'); t1.className = 'ui-title'; t1.textContent = 'Meal Services';
    const row1 = document.createElement('div'); row1.className = 'hstack';
    const couponBtn = document.createElement('button'); couponBtn.className = 'btn'; couponBtn.textContent = state.flags.mealCoupon ? 'Meal Coupon Collected' : 'Get Meal Coupon';
    couponBtn.disabled = !!state.flags.mealCoupon;
    couponBtn.addEventListener('click', function(){ if(state.flags.mealCoupon) return; state.flags.mealCoupon = true; addItem('Meal Coupon'); couponBtn.textContent = 'Meal Coupon Collected'; couponBtn.disabled = true; });
    row1.appendChild(couponBtn);
    services.appendChild(t1); services.appendChild(row1);

    const dining = document.createElement('div');
    dining.className = 'ui-card';
    const t2 = document.createElement('div'); t2.className = 'ui-title'; t2.textContent = 'Dining';
    const row2 = document.createElement('div'); row2.className = 'hstack';
    const eatBtn = document.createElement('button'); eatBtn.className = 'btn primary'; eatBtn.textContent = state.flags.ateMeal ? 'Meal Eaten' : 'Eat Meal (+5 XP, Energy 100)';
    eatBtn.disabled = !!state.flags.ateMeal || !state.flags.mealCoupon;
    eatBtn.addEventListener('click', function(){ if (state.flags.ateMeal || !state.flags.mealCoupon) return; state.flags.ateMeal = true; setEnergy(100); addXP(5); completeQuest('eatMeal'); eatBtn.textContent = 'Meal Eaten'; eatBtn.disabled = true; });
    row2.appendChild(eatBtn);
    dining.appendChild(t2); dining.appendChild(row2);

    root.appendChild(services);
    root.appendChild(dining);
    container.appendChild(root);
  };
})();
